### ISSUE TITLE
Gson deserialization fails for 'Quantity' using GraalVM native-image

### ISSUE THREAD

**Issue by eschrock** [kubernetes-client/java#809]
We are using kubernetes-client with GraalVM [native-image](https://www.graalvm.org/docs/reference-manual/native-image/). Most of it works just fine with the proper reflect config, but efforts to get status fail:

```
Unable to invoke no-args constructor for class io.kubernetes.client.custom.Quantity$QuantityAdapter. Registering an InstanceCreator with Gson for this type may fix this problem.
java.lang.RuntimeException: Unable to invoke no-args constructor for class io.kubernetes.client.custom.Quantity$QuantityAdapter. Registering an InstanceCreator with Gson for this type may fix this problem.
	at com.google.gson.internal.ConstructorConstructor$14.construct(ConstructorConstructor.java:228)
	at com.google.gson.internal.bind.JsonAdapterAnnotationTypeAdapterFactory.getTypeAdapter(JsonAdapterAnnotationTypeAdapterFactory.java:55)
	at com.google.gson.internal.bind.JsonAdapterAnnotationTypeAdapterFactory.create(JsonAdapterAnnotationTypeAdapterFactory.java:49)
...
	at com.google.gson.Gson.getAdapter(Gson.java:458)
	at com.google.gson.Gson.fromJson(Gson.java:931)
	at com.google.gson.Gson.fromJson(Gson.java:897)
	at com.google.gson.Gson.fromJson(Gson.java:846)
	at io.kubernetes.client.JSON.deserialize(JSON.java:108)
	at io.kubernetes.client.ApiClient.deserialize(ApiClient.java:668)
	at io.kubernetes.client.ApiClient.handleResponse(ApiClient.java:871)
	at io.kubernetes.client.ApiClient.execute(ApiClient.java:798)
	at io.kubernetes.client.apis.AppsV1Api.readNamespacedStatefulSetWithHttpInfo(AppsV1Api.java:7877)
	at io.kubernetes.client.apis.AppsV1Api.readNamespacedStatefulSet(AppsV1Api.java:7859)
Caused by: java.lang.IllegalArgumentException: Class io.kubernetes.client.custom.Quantity$QuantityAdapter is instantiated reflectively but was never registered. Register the class by using org.graalvm.nativeimage.hosted.RuntimeReflection
	at com.oracle.svm.core.genscavenge.graal.AllocationSnippets.checkDynamicHub(AllocationSnippets.java:163)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at com.google.gson.internal.UnsafeAllocator$1.newInstance(UnsafeAllocator.java:50)
	at com.google.gson.internal.ConstructorConstructor$14.construct(ConstructorConstructor.java:225)
	... 51 more
```

Quantity does declare a type adapter (like IntOrString), so it should work that I can use a similar reflection config:

```
{
  "name":"io.kubernetes.client.custom.IntOrString$IntOrStringAdapter",
  "methods":[{"name":"<init>","parameterTypes":[] }]
},
```

However, doing this fails during the image build process:

```
Method io.kubernetes.client.custom.Quantity$QuantityAdapter.<init>() not found
```

I believe the issue here is that `Quantity::QuantityAdapter` is a non-static class, unlike `IntOrString::IntOrStringAdapter`. My best guess is that gson can deal with this through unsafe operations, but we can't. I will add a comment shortly with a test program and (hopefully) a fix if the static-ness is the only issue.

**Comment by eschrock** [comment]
I put together a [quick example](https://github.com/kubernetes-client/java/files/3880861/app.zip) to demonstrate. After running `mvn package` with GraalVM 19.2.0:

```
$ java -jar target/app-1.0-SNAPSHOT-jar-with-dependencies.jar 
intOrString = 4
quanitity = 1Gi
$ ./target/io.titandata.app.app 
intOrString = 4
Exception in thread "main" java.lang.RuntimeException: Unable to invoke no-args constructor for class io.kubernetes.client.custom.Quantity$QuantityAdapter. Register an InstanceCreator with Gson for this type may fix this problem.
	at com.google.gson.internal.ConstructorConstructor$14.construct(ConstructorConstructor.java:226)
	at com.google.gson.internal.bind.JsonAdapterAnnotationTypeAdapterFactory.getTypeAdapter(JsonAdapterAnnotationTypeAdapterFactory.java:55)
	at com.google.gson.internal.bind.JsonAdapterAnnotationTypeAdapterFactory.create(JsonAdapterAnnotationTypeAdapterFactory.java:49)
	at com.google.gson.Gson.getAdapter(Gson.java:423)
	at com.google.gson.Gson.fromJson(Gson.java:886)
	at com.google.gson.Gson.fromJson(Gson.java:852)
	at com.google.gson.Gson.fromJson(Gson.java:801)
	at com.google.gson.Gson.fromJson(Gson.java:773)
	at io.titandata.app.App.main(App.java:19)
Caused by: java.lang.IllegalArgumentException: Class io.kubernetes.client.custom.Quantity$QuantityAdapter is instantiated reflectively but was never registered. Register the class by using org.graalvm.nativeimage.hosted.RuntimeReflection
	at com.oracle.svm.core.genscavenge.graal.AllocationSnippets.checkDynamicHub(AllocationSnippets.java:163)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at com.google.gson.internal.UnsafeAllocator$1.newInstance(UnsafeAllocator.java:50)
	at com.google.gson.internal.ConstructorConstructor$14.construct(ConstructorConstructor.java:223)
	... 8 more
```

Trying to add a similar declaration to the reflect-config file generates the previously mentioned error:

```
Error: Error parsing reflection configuration in jar:file:/Users/eric.schrock/dev/client-bug/app/target/app-1.0-SNAPSHOT.jar!/reflect-config.json:
Class io.kubernetes.client.custom.Quanitity$QuantityAdapter not found
```

**Comment by brendandburns** [comment]
@eschrock I think this can be closed since #810 merged?

**Comment by eschrock** [comment]
Yep!
